core/commands/storage.go

```diff
@@ -375,6 +375,65 @@ func controlSessionTimeout(ss *storage.FileContracts, stateFlow []*storage.FlowC
 	}
 }
 
+type paramsForPrepareContractsForShard struct {
+	ctx       context.Context
+	rq        *storage.RetryQueue
+	api       coreiface.CoreAPI
+	ss        *storage.FileContracts
+	n         *core.IpfsNode
+	test      bool
+	isRepair  bool
+	renterPid string
+	shardKey  string
+	shard     *storage.Shard
+}
+
+func prepareSignedContractsForShard(param *paramsForPrepareContractsForShard, candidateHost *storage.HostNode) error {
+	shard := param.shard
+	shardIndex := shard.ShardIndex
+
+	shard.SetPrice(candidateHost.Price)
+	cfg, err := param.n.Repo.Config()
+	if err != nil {
+		return err
+	}
+
+	// init escrow Contract
+	_, pid, err := ParsePeerParam(candidateHost.Identity)
+	if err != nil {
+		return err
+	}
+	escrowContract, err := escrow.NewContract(cfg, shard.ContractID, param.n, pid, shard.TotalPay)
+	if err != nil {
+		log.Error("create escrow contract failed. ", err)
+		return err
```

Comment:
Really small improvement: can you make this return `fmt.Errorf("create escrow contracted failed: [%v]", err)` for all returns in this function and remove `log.Error`, otherwise it will be double logging. The `sendStepStateChan` will log automatically. Thanks.